build/docker-compose.yaml

```diff
@@ -61,12 +61,20 @@ services:
 
   snapshot-ctest: &snapshot-ctest
     <<: *build-setup
-    command: bash -c 'if [ -f CMakeLists.txt ]; then mkdir -p work && cd work && cmake .. && make -j "$${MAKEJOBS}" && ctest; fi'
+    command: bash -c 'if [ -f CMakeLists.txt ]; then mkdir -p work && cd work && cmake .. && make -j "$${MAKEJOBS}" && ctest -L fast -j "$${MAKEJOBS}" --output-on-failure; fi'
 
   prb-ctest:
     <<: *snapshot-ctest
 
 
+  snapshot-correctness: &snapshot-correctness
+    <<: *build-setup
+    command: bash -c 'if [ -f CMakeLists.txt ]; then mkdir -p work && cd work && cmake .. && make -j "$${MAKEJOBS}" && ctest -j "$${MAKEJOBS}" --output-on-failure; fi'
```

Comment:
I don't like the fact that this puts the build directory into the source directory for two reasons:

1. On some systems (for example Fedora), docker needs to run as root. Having files created as another user in your build directory is annoying.
1. Most people in my team work locally on their notebooks and use `rsync` to put their code to a remote machine. Having files on one machine but not on the other makes stuff generally harder.

If you look at #1141 I proposed a different `docker-compose` that can use a build directory anywhere. It just uses whatever it finds in the environment variable `BUILDDIR`. This is imho more flexible and it wouldn't interfere that much with our workflow